clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/agent/OpenstackLoadBalancerCachingAgent.groovy

```diff
@@ -107,8 +107,8 @@ class OpenstackLoadBalancerCachingAgent extends AbstractOpenstackCachingAgent {
       Collection<CacheData> serverGroupsData = providerCache.getAll(SERVER_GROUPS.ns, filters, RelationshipCacheFilter.include(LOAD_BALANCERS.ns))
       Set<OpenstackServerGroup> serverGroups = serverGroupsData?.findAll { s ->
         s.relationships[LOAD_BALANCERS.ns].find { lbKey -> lbKey == loadBalancerKey } != null
-      }?.collect { s ->
-        objectMapper.convertValue(s?.attributes, OpenstackServerGroup)
+      }?.findResults { s ->
+        objectMapper.convertValue(s.attributes, OpenstackServerGroup)
```

Comment:
I thought `findResults` prunes `null` values returned from the closure, not passed to the closure. I think you still need `s?.attributes` here.